### PR TITLE
doc(MCRDMPE-313) - Add documentation

### PR DESCRIPTION
**Pull request by alexBonadei**

### CODE REVIEW

File: docs/20_Walkthrough.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+- The `<type>` part is related to the type of the implementation. The possible options are: feat, fix and doc;
+- The `<implementation-task-id>` is the identifier of the implementation task. Usually, it is the Jira story code;
+- The `<description>` is the description of the story. It should be a short recap of the story.
+
+Moreover, for the same traceability requirement, also the commit messages should follow a specific convention, that is the following: `[<implementation-task-id>] <commit-description>`. Example: `[MC-1] Added login endpoint`.
+- The `<implementation-task-id>` is the identifier of the implementation task. Usually, it is the Jira story code;
+- The `<commit-description>` is the description of the commit.
+
+If the branch has been named correctly, the implementation task identifier is inserted automatically in the commit message. Therefore, when you run the command `git commit -m "message"`, the commit message is automatically created as `[<implementation-task-id>] message`. The automatic insertion of the implementation task identifier is performed by [jira-prepare-commit-msg](https://github.com/bk201-/jira-prepare-commit-msg).
```
Comment (calairoh) [inline docs/20_Walkthrough.md]: @alexBonadei Are you sure that the commit message insertion is performed by `jira-prepare-commit-msg`? As far as I know that library is for NodeJS.

File: docs/20_Walkthrough.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+You can customize the branch naming convention by modifying the bash script that checks the branch name. In particular, you need to modify the regex in the `./scripts/check-branch-naming.sh` with your convention. You can also disable the check by removing the run of the `check-branch-naming.sh` bash script directly in the `./.husky/pre-commit` file.
+
+Moreover, you can remove the automatic insertion of the implementation task identifier by removing the run of the `npx jira-prepare-commit-msg` command directly in the `./.husky/prepare-commit-msg` file.
```
Comment (calairoh) [inline docs/20_Walkthrough.md]: Same here. The way the commit is created is different from the one used in NodeJS template. I think this part has to be fixed.

**Review by calairoh — CHANGES_REQUESTED** [review]
PR LGTM
There is only a fix to do in the GitOps section regarding the way the template works for commint and branch automation

File: docs/20_Walkthrough.md (hunk 1 above)
Comment (alexBonadei) [reply docs/20_Walkthrough.md]: Actually I implemented that script for the commit message insertion lol

File: docs/20_Walkthrough.md (hunk 2 above)
Comment (alexBonadei) [reply docs/20_Walkthrough.md]: Yep

**Review by calairoh — APPROVED** [review]
LGTM

**Pull request merged**